src/Arcus.Observability.Tests.Integration/Serilog/ApplicationInsightsSinkTests.cs

```diff
@@ -237,43 +322,66 @@ public void LogSqlDependency_SinksToApplicationInsights_ResultsInSqlDependencyTe
+                {
+                    EventsResults<EventsDependencyResult> results = await client.GetDependencyEventsAsync();
+                    Assert.NotEmpty(results.Value);
+                    Assert.Contains(results.Value, result =>
+                    {
+                        return result.Dependency.Type == "SQL"
+                               && result.Dependency.Target == serverName
+                               && result.Dependency.Name == $"SQL: {databaseName}/{tableName}";
+                    });
+                });
             }
         }
 
         [Fact]
-        public void LogEventWithComponentName_SinksToApplicationInsights_ResultsInTelemetryWithComponentName()
+        public async Task LogEventWithComponentName_SinksToApplicationInsights_ResultsInTelemetryWithComponentName()
         {
             // Arrange
+            string message = _bogusGenerator.Lorem.Sentence();
             string componentName = _bogusGenerator.Commerce.ProductName();
             using (ILoggerFactory loggerFactory = CreateLoggerFactory(config => config.Enrich.WithComponentName(componentName)))
             {
                 ILogger logger = loggerFactory.CreateLogger<ApplicationInsightsSinkTests>();
 
                 // Act
-                logger.LogInformation("This message will be enriched with a component name");
+                logger.LogInformation(message);
+            }
 
-                // Assert
-                // Hold on till we have agreed on assertion...
+            // Assert
+            using (ApplicationInsightsDataClient client = CreateApplicationInsightsClient())
+            {
+                await RetryAssertUntilTelemetryShouldBeAvailableAsync(async () =>
+                {
+                    EventsResults<EventsTraceResult> results = await client.GetTraceEventsAsync(filter: OnlyLastHourFilter);
+                    Assert.NotEmpty(results.Value);
+                    Assert.Contains(results.Value, result => result.Trace.Message == message && result.Cloud.RoleName == componentName);
+                });
             }
         }
 
         [Fact]
-        public void LogEventWithVersion_SinksToApplicationInsights_ResultsInTelemetryWithVersion()
+        public async Task LogEventWithVersion_SinksToApplicationInsights_ResultsInTelemetryWithVersion()
         {
             // Arrange
+            var eventName = "Update version";
             using (ILoggerFactory loggerFactory = CreateLoggerFactory(config => config.Enrich.WithVersion()))
             {
                 ILogger logger = loggerFactory.CreateLogger<ApplicationInsightsSinkTests>();
 
                 // Act
-                logger.LogEvent("Update version");
-
-                // Assert
-                // Hold on till we have agreed on assertion...
+                logger.LogEvent(eventName);
             }
+
+            // Assert
+            // Hold on till we have agreed on assertion...
```

Comment:
Is this on purpose that we don't cover it?

Reply:
As I previously told you on Teams, the problem I had is that the retrieved entries doesn't contain (yet, it's a preview NuGet package) custom properties, so also no enriched properties. What I'm thinking of what we could do, is altering the message template with a test enricher if the log event got enriched with something we want to test.

Reply:
I think it's ok how it is in that case, resolving!